db/posts/procedures.sql

```diff
@@ -45,39 +45,51 @@ END $$
 DELIMITER ;
 
 -- call as: CALL delete_post(1);
+
 DROP PROCEDURE IF EXISTS get_all_posts;
 DELIMITER $$
+
 CREATE PROCEDURE get_all_posts()
 BEGIN
   SELECT 
     posts.*,
-    users.username AS username,
-    users.email AS email
+    users.username,
+    users.email,
+    COALESCE(COUNT(pr.post_id), 0) AS total_reactions
```

Comment:
there is not need for a COALESCE because the inner join already return a 0 any row match the requirements